include/libdnf5/rpm/versionlock_config.hpp

```diff
@@ -0,0 +1,144 @@
+    libdnf5::sack::QueryCmp get_comparator() const { return comparator; }
+    /// Get the value.
+    std::string get_value() const { return value; }
+
+    /// Get the key as a string.
+    std::string get_key_str() const { return key_str; }
+    /// Get the comparison operator as a string.
+    std::string get_comparator_str() const { return comparator_str; }
+
+    /// Get list of errors found during parsing the entry from configuration file.
+    std::vector<std::string> get_errors() const { return errors; }
+
+    /// Converts the condition to "key operator value" string usable for printing.
+    /// @param with_errors Include also error messages for invalid entries
+    std::string to_string(bool with_errors) const;
+
+private:
+    // map string comparator to query cmp_type
+    static const std::map<std::string, libdnf5::sack::QueryCmp> VALID_COMPARATORS;
+    bool valid;
+    std::string key_str;
+    Keys key;
+    std::string comparator_str;
+    libdnf5::sack::QueryCmp comparator{0};
+    std::string value;
+    std::vector<std::string> errors{};
+};
+
+/// One versionlock configuration file entry. It consists of the
+/// package name and a set of conditions. All conditions must be true
+/// for package version to get locked.
+class VersionlockPackage {
+public:
+    /// Creates an instance of `VersionlockPackage` class specifying the
+    /// name of package.
+    /// @param name Name of the package to be configured
+    VersionlockPackage(std::string_view name, std::vector<libdnf5::rpm::VersionlockCondition> && conditions);
+
+    /// Returns true if this configuration entry is valid.
+    bool is_valid() const { return valid; }
+
+    /// Get the package name.
+    std::string get_name() const { return name; }
+
+    /// Get the comment for this entry.
+    std::string get_comment() const { return comment; }
+    /// Set comment for this entry.
+    void set_comment(std::string_view comment);
+
+    /// Get the list of conditions configured for the package.
+    std::vector<VersionlockCondition> get_conditions() const { return conditions; }
+
+    /// Add a new condition for the package
+    void add_condition(VersionlockCondition && condition);
+
+    /// Get list of errors found during parsing the entry from configuration file.
+    std::vector<std::string> get_errors() const { return errors; }
+
+    /// Converts the package configuration to string usable for printing.
+    /// @param with_errors Include also error messages for invalid entries
+    std::string to_string(bool with_errors, bool with_comment) const;
+
+private:
+    bool valid;
+    std::string name;
+    std::string comment;
+    std::vector<VersionlockCondition> conditions{};
+    std::vector<std::string> errors{};
+};
+
+
+/// Class contains parsed versionlock configuration file.
+class VersionlockConfig {
+public:
+    /// Get list of configured versionlock entries.
+    std::vector<VersionlockPackage> & get_packages() { return packages; }
```

Comment:
`const std::vector<VersionlockPackage> &`?

Reply:
Not that easily. Given that returned reference is used to directly manipulate the vector (add new items, delete some items, even clear the vector completely), it would require more API on VersionlockPackage class.